backend/src/packages/chaiNNer_standard/image_adjustment/adjustments/premultiplied_alpha.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from enum import Enum
+
+import numpy as np
+
+from nodes.impl.image_utils import as_target_channels
+from nodes.properties.inputs import EnumInput, ImageInput
+from nodes.properties.outputs import ImageOutput
+
+from .. import adjustments_group
+
+
+class AlphaAssociation(Enum):
+    UNPREMULTIPLY_RGB = "Unpremultiply RGB"
+    PREMULTIPLY_RGB = "Premulitply RGB"
+
+
+@adjustments_group.register(
+    schema_id="chainner:image:premultiplied_alpha",
+    description="Divide (or multiply) the RGB channels with the Alpha channel values of an image to go from a Straight Alpha to a Premultiplied Alpha image state.",
+    name="Premultiplied Alpha",
+    icon="CgMathDivide",
+    inputs=[
+        ImageInput(),
+        EnumInput(AlphaAssociation, label="Alpha Association"),
+    ],
+    outputs=[ImageOutput(image_type="Input0")],
+)
+def premultiplied_alpha_node(
+    img: np.ndarray, alpha_association: AlphaAssociation
+) -> np.ndarray:
+    if len(img.shape) != 3:
+        return img, np.ones(img.shape[:2], dtype="float32")
```

Comment:
This is the reason for the type error. What is this even supposed to do?

Reply:
Originally the thought was that if the incoming image only had 3 channels we would pad to 4 channels with a solid channel of 1's. I'll just always force a 4 channel input and this can go away.